### ISSUE TITLE
Segmentation Fault on Startup

### ISSUE THREAD

**Issue by 99MatthewBarton** [hyprwm/Hyprland#2709]
**STEPS TO REPRODUCE**
Login from display manager.

**EXPECTED OUTCOME**
For the past few days of using Hyprland I have been able to login normally.

**OUTCOME:**
A black screen appears (the normal Hyprland startup since I have the logo and splash disabled), then returns to the display manager. Crash report indicates this is segmentation fault.

**ATTEMPTED FIXES/NOTES**
Did obvious things like rebooting and pacman -Syu.
I have tried reinstalling Hyprland, launching with a default config file and downgrading to previous installed version.
Login through i3 works normally.
I might be missing something obvious (I hope not). I'm fairly new to Linux and only started using Hyprland a few days ago.
System is a Dell XPS 13 9300 with Intel i5-1035G1 CPU, integrated graphics.

**LOGS/CONFIG**
[crashreport.txt](https://github.com/hyprwm/Hyprland/files/12048250/crashreport.txt)
[hyprland.txt](https://github.com/hyprwm/Hyprland/files/12048224/hyprland.txt)
[coredump.txt](https://github.com/hyprwm/Hyprland/files/12048251/coredump.txt)

**Comment by vaxerski** [comment]
ca you `git bisect` if this is a regression?

**Comment by 99MatthewBarton** [comment]
I don't have a known good version, when I reverted Hyprland, the issue persisted. That makes me think it's a conflict somewhere but the logs don't give me much direction to investigate that.

**Comment by vaxerski** [comment]
[patch.txt](https://github.com/hyprwm/Hyprland/files/12059083/patch.txt)
try this

**Comment by 99MatthewBarton** [comment]
I'm sorry but I can't figure out how to apply this patch.

**Comment by MightyPlaza** [comment]
git apply

**Comment by 99MatthewBarton** [comment]
I'm getting:
```
> error: src/managers/AnimationManager.cpp: No such file or directory
> error: subprojects/wlroots: No such file or directory
```

**Comment by 99MatthewBarton** [comment]
I've also tried uninstalling the packages that had been installed since Hyprland, but that didn't affect the outcome.

**Comment by MightyPlaza** [comment]
you probably are in the wrong dir, make sure you're in Hyprland named dir

**Comment by 99MatthewBarton** [comment]
Same issue in /usr/include/hyprland and /usr/share/hyprland.

**Comment by MightyPlaza** [comment]
![image](https://github.com/hyprwm/Hyprland/assets/123664421/4b12dce0-0954-4beb-9d51-e8abe6a7e6ff)

**Comment by 99MatthewBarton** [comment]
Thanks for you patience. When I run Hyprland in i3 it launches. Sorry I didn't check that earlier. When I do that I get:
`00:00:00.051 [types/wlr_drm_lease_v1.c:715] No DRM backend supplied, failed to create wlr_drm_lease_v1_manager
00:00:00.051 [backend/headless/backend.c:68] Creating headless backend
00:00:00.062 [backend/x11/backend.c:164] Starting X11 backend
00:00:00.068 [backend/headless/backend.c:16] Starting headless backend
[196 12:53:13.002070] [glfw error 65544]: process_desktop_settings: failed with error: [org.freedesktop.DBus.Error.ServiceUnknown] The name org.freedesktop.portal.Desktop was not provided by any .service files
`
I can't find the Hyprland directory for some reason. When I find from root, it only shows me directories in usr/include and usr/share.

**Comment by 99MatthewBarton** [comment]
I can launch normally into hyprland again. I have no idea what changed. The only thing I did between this reset and the last was run hyprland from terminal in i3. Thanks for your help and patience. I'll mark as closed.

**Comment by 99MatthewBarton** [comment]
I'm still getting this issue inconsistently whenever I restart hyprland. Still aren't closer to pinpointing a solution.

**Comment by vaxerski** [comment]
[patch.txt](https://github.com/hyprwm/Hyprland/files/12094605/patch.txt)

try this patch

**Comment by 99MatthewBarton** [comment]
Do you know where the pacman default Hyprland dir is?

**Comment by vaxerski** [comment]
you'll have to build manually from git

**Comment by 99MatthewBarton** [comment]
Ah alright, I'll get back to you when I've tried that.

**Comment by pranavrao145** [comment]
I'm on the latest commit (which at this point is 1ec0b7b59a84a07f97c3ebee9f6694bc4a1b1dcd) and I'm also experiencing this issue with SDDM. I did a `git bisect` and found this wlroots update  (8c9e2e1ff1d22f7954f5e75194e9f77cacffa3f6) to be the problem. I can't apply the patches because I have no idea what commit this issue was opened on, but this info might be useful regardless.

**Comment by dan-cooke** [comment]
I have this problem on 0.28 and there is an issue cloning the  wlroots subrepo via `yay`

The git server is returning a 503.

**Comment by memchr** [comment]
> I have this problem on 0.28 (pacman latest binary) and there is an issue cloning the wlroots subrepo via `yay`
``

I do not think it is appropriate to ask irrelevant questions in GitHub issues. If you come across a problem, please post your inquiry in the GitHub discussion or on Discord.

**gitlab.freedesktop.org** is currently unavailable from as the registry data is being migrated from Google Cloud to a self-hosted solution. They estimate it will be between 8 and 12 UTC. However, they seem to have run into some sort of problem and have extended the downtime as it is now 17 UTC.

**Comment by dan-cooke** [comment]
@memchr I'm sorry but where exactly did I ask a question? I was reporting an issue that I was encountering while trying to install this library.

Last time I checked - that is what github issues is for.

**Comment by memchr** [comment]
> @memchr I'm sorry but where exactly did I ask a question? I was reporting an issue that I was encountering while trying to install this package.
> 
> Last time I checked - that is what github issues is for.

Could you please double check the current thread you are posting to? #2709 is `Segmentation Fault on Startup`. I mean you no offense, but your question is not related to this issue in anyway. If you wish to report an bug, please create a new issue thread rather than randomly commenting on an existing unrelated issue.

**Comment by dan-cooke** [comment]
@memchr The first thing I said in my comment is that I am encountering this issue on 0.28 of `Hyprland`

My issue is almost word-for-word what OP said. So I thought I would just give this thread some traction.

Apologies if this is not allowed

**Comment by memchr** [comment]
> @memchr The first thing I said in my comment is that I am encountering this issue on 0.28 of `Hyprland`
> 
> > I have this problem on 0.28 (pacman latest binary)
> 
> My issue is almost word-for-word what OP said. So I thought I would just give this thread some traction.
> 
> Apologies if this is not allowed

No, it is not my position to say that it is not allowed, I am just recommending that you create a new issue thread instead of posting to a completely unrelated existing issue, thus confusing people tracking this bug.

Please do a double check of the original post: 

> **STEPS TO REPRODUCE** Login from display manager.
> 
> **EXPECTED OUTCOME** For the past few days of using Hyprland I have been able to login normally.
> 
> **OUTCOME:** A black screen appears (the normal Hyprland startup since I have the logo and splash disabled), then returns to the display manager. Crash report indicates this is segmentation fault.
> 
> **ATTEMPTED FIXES/NOTES** Did obvious things like rebooting and pacman -Syu. I have tried reinstalling Hyprland, launching with a default config file and downgrading to previous installed version. Login through i3 works normally. I might be missing something obvious (I hope not). I'm fairly new to Linux and only started using Hyprland a few days ago. System is a Dell XPS 13 9300 with Intel i5-1035G1 CPU, integrated graphics.

Sorry if you are confused by the github's user interface, but the OP's problem is clearly completely unrelated to yours.

Please do not reply if possible.

**Comment by dan-cooke** [comment]
@memchr just to clarify, as I see now why you might think I am posting in an unrelated thread,

I am encountering this issue that OP is encountering, when trying to boot `Hyprland` from `sddm`  

```
Aug 27 12:16:40 danarch sddm[744]: Authentication error: SDDM::Auth::ERROR_INTERNAL "Process crashed"
Aug 27 12:16:40 danarch sddm[744]: Auth: sddm-helper (--socket /tmp/sddm-auth-0197350e-a93c-41c6-803b-7ef0c33313e8 --id 1 --start /usr/local/bin/wrappedhl --user dan) crashed (exit code 1)
Aug 27 12:16:40 danarch sddm[744]: Authentication error: SDDM::Auth::ERROR_INTERNAL "Process crashed"
```

Aside from that my hyprland crash report is identical to OP's as I already stated, as is my coredump after the crash.

The issue I reported about freedesktop was just an aside. 

Let's not be rude please

**Comment by 99MatthewBarton** [comment]
I have eventually isolated the issue. The seg fault only occurs when my laptop is charging. I have no clue what the connection is, but it's consistently reproducible.

**Comment by romanstingler** [comment]
@99MatthewBarton are you still using hyprland? 
have you installed the latest bios and firmware files for your system?? 
have you tried to build the latest version?  
in AUR you have hyprland-git package

**Comment by 99MatthewBarton** [comment]
@romanstingler yes I'm still using hyprland. I was on hyprland-git for a while but I moved back to the stable branch. The issue still persists. It is definitely caused by having the charging cable plugged in so it is easy to bypass.

**Comment by vaxerski** [comment]
Hello there!

This issue has been closed, as we are moving from issues to discussions. More about this can be found [here](https://wiki.hyprland.org/Contributing-and-Debugging/Issue-Guidelines/) and in #9854.

Firstly, please make sure this issue is still relevant in latest Hyprland. If so, we ask you to open a [discussion](https://github.com/hyprwm/Hyprland/discussions) (please read the discussion guidelines first, in the pinned post)

Thank you for your understanding!
